src/jaegertracing/ThriftMethods.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jaegertracing/thrift-gen/BaggageRestrictionManager.h"
+#include "jaegertracing/thrift-gen/sampling_types.h"
+
+namespace jaegertracing {
+
+// trivial constructors of BaggageRestrictionManager thrift generated code are not generated by MSVC
```

Comment:
What?! Why not? This is why I am not so keen on supporting Windows.

Reply:
I didnt dig a lot. For example, you can see that the copy ctor of BaggageRestrictionManager_getBaggageRestrictions_args is declared in the header but isnt defined.

Reply:
How can we proceed on this point? I think we can follow the bug in parallel, unless the thrift will dropped for protobuf.

Reply:
There is no issue. You are correct. See if this works without the `#ifdef _MSVC`. It might not just be a Windows issue.